syntax = "proto3";

import "ls_state.proto";

package bayesmix;

message ClusterVal {
    oneof val {
      UnivLSState univ_ls_state = 1;
      MultiLSState multi_ls_state = 2;
    }
  }


message SemiHdpState {
  message RestaurantState {
      repeated ClusterVal theta_stars = 1;
      repeated int32 n_by_clus = 2;
      repeated int32 table_to_shared = 3;
      repeated int32 table_to_idio = 4;
  }

  message GroupState {
      repeated int32 cluster_allocs = 1;
  }

  repeated RestaurantState restaurants = 1;
  repeated GroupState groups = 2;
  repeated ClusterVal taus = 3;
  repeated int32 c = 4;
  double w = 5;
}